app/ide-desktop/lib/dashboard/src/authentication/src/dashboard/components/templates.tsx

```diff
@@ -1,62 +1,137 @@
 /** @file Renders the list of templates that can be used to create a project from. */
 // eslint-disable-next-line @typescript-eslint/no-unused-vars
-import * as react from "react";
+import * as React from 'react'
 
-const templates = [
-  { title: "New empty project", templateName: undefined },
-  { title: "Colordo COVID", templateName: "Colorado_COVID" },
-  { title: "KMeans", templateName: "Kmeans" },
-  { title: "NASDAQ Returns", templateName: "NASDAQ_Returns" },
-  { title: "Restaurants", templateName: "Orders" },
-  { title: "Github Stars", templateName: "Stargazers" },
-];
+// =================
+// === Templates ===
+// =================
 
-interface Props {
-  onChange: (templateName: string | undefined) => void;
+/** Define the template type to render */
+interface Template {
+    title: string
+    description?: string
```

Comment:
i think it might be better to make `description` and `templateName` required, and just make them empty strings (`''`) for the empty project template...

... you *are* doing `template.templateName ? A : B` later down so i get why it's optional, but `""` is also falsy so it should still work

Reply:
alternatively, you can check whether `template == emptyTemplate` (and keep the `emptyTemplate` variable) instead

Reply:
The templateName is the template name that needs to be passed to the backend.
Passing a null value may cause unexpected behavior, so I cannot guarantee the outcome.

Reply:
you mean passing an empty string? makes sense, it might be good (but not necessary) to add a comment explaining why they should be optional instead of using the empty string